pkg/lib/config/secret_update_instruction.go

```diff
@@ -569,9 +577,66 @@ func (i *BotProtectionProviderCredentialsUpdateInstruction) unset(currentConfig
 	return out, nil
 }
 
+type SAMLIdpSigningSecretsUpdateInstruction struct {
+	Action SecretUpdateInstructionAction `json:"action,omitempty"`
+}
+
+func (i *SAMLIdpSigningSecretsUpdateInstruction) ApplyTo(ctx *SecretConfigUpdateInstructionContext, currentConfig *SecretConfig) (*SecretConfig, error) {
+	switch i.Action {
+	case SecretUpdateInstructionActionGenerate:
+		return i.generate(currentConfig)
+	default:
+		return nil, fmt.Errorf("config: unexpected action for SAMLIDPSigningSecretsUpdateInstruction: %s", i.Action)
+	}
+}
+
+func (i *SAMLIdpSigningSecretsUpdateInstruction) generate(currentConfig *SecretConfig) (*SecretConfig, error) {
+	out := &SecretConfig{}
+	var credentials *SAMLIdpSigningMaterials
+	for _, item := range currentConfig.Secrets {
+		if item.Key == SAMLIdpSigningMaterialsKey {
+			credentials = item.Data.(*SAMLIdpSigningMaterials)
+		}
+		out.Secrets = append(out.Secrets, item)
+	}
+
+	if credentials == nil {
+		credentials = &SAMLIdpSigningMaterials{
+			Certificates: []*SAMLIdpSigningCertificate{},
+		}
+	}
+
+	// FIXME: The common name is not important, but we might want to put the app id here.
+	newCert, err := GenerateSAMLIdpSigningCertificate("urn:authgear.com")
```

Comment:
What value did we use previously?

Reply:
For existing projects, I created the certs manually with a value like `urn:project_id.authgear.com`